ts/features/bonus/cgn/components/otp/OtpCodeComponent.tsx

```diff
@@ -0,0 +1,250 @@
+        rt.minutes === 0
+          ? ""
+          : I18n.t("bonus.cgn.otp.code.minutes", {
+              defaultValue: I18n.t("bonus.cgn.otp.code.minutes.other", {
+                minutes: rt.minutes
+              }),
+              count: rt.minutes,
+              minutes: rt.minutes
+            }) + " ";
+      const seconds = I18n.t("bonus.cgn.otp.code.seconds", {
+        defaultValue: I18n.t("bonus.cgn.otp.code.seconds.other", {
+          seconds: rt.seconds
+        }),
+        count: rt.seconds,
+        seconds: rt.seconds
+      });
+      return `${I18n.t("bonus.cgn.otp.code.validUntil")} ${minutes}${seconds}`;
+    })
+    .getOrElse(undefined);
+
+export const OtpCodeComponent = (props: Props) => {
+  const { startPercentage, endPercentage } = props.progressConfig ?? {
+    startPercentage: 0,
+    endPercentage: 100
+  };
+  const [progressWidth] = useState(new Animated.Value(startPercentage));
+  const [elapsedSeconds, setElapsedSeconds] = useState(0);
+  const [duration, setDuration] = useState<Second>(0 as Second);
+  const [expirationTime, setExpirationTime] = useState<
+    ExpirationTime | undefined
+  >(undefined);
+  const intervalRef = useRef<undefined | number>(undefined);
+  const formattedCode = addEvery(props.otp.code, " ", 3);
+
+  // reset interval timer
+  const stopInterval = () => clearInterval(intervalRef.current);
+
+  /**
+   * when otp changes
+   * - reset elapsed seconds
+   * - reset progress width
+   * - start progress width animation
+   */
+  useEffect(() => {
+    const currentDuration = getOtpTTL(props.otp);
+    setDuration((currentDuration / 1000) as Second);
+    setElapsedSeconds(0);
+    progressWidth.setValue(startPercentage);
+
+    Animated.timing(progressWidth, {
+      useNativeDriver: false,
+      toValue: endPercentage,
+      duration: currentDuration as number,
+      easing: Easing.linear
+    }).start(() => {
+      // when animation end
+      setElapsedSeconds(cv => cv + 1);
+      stopInterval();
+      props.onEnd();
+    });
+    // eslint-disable-next-line functional/immutable-data
+    intervalRef.current = setInterval(() => {
+      setElapsedSeconds(cv => cv + 1);
+    }, 1000);
+    return stopInterval;
+  }, [props.otp.code]);
+
+  // update the remaining time
+  useEffect(() => {
+    if (duration === 0) {
+      return;
+    }
+    const expiration = duration - elapsedSeconds;
+    const minutes = Math.max(0, Math.floor(expiration / 60));
+    const seconds = Math.max(0, Math.floor(expiration % 60));
+    setExpirationTime({ minutes, seconds });
+  }, [duration, elapsedSeconds]);
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.remainingTimeContainer}>
+        <Label weight={"Regular"} color={"bluegrey"}>
+          {I18n.t("bonus.cgn.otp.code.title")}
+        </Label>
+      </View>
+      <TouchableDefaultOpacity
+        style={styles.otpContainer}
+        onPress={() => clipboardSetStringWithFeedback(formattedCode)}
```

Comment:
I guess maybe would be better to copy the raw format of the code and not the spaced one used only for a representation purpose
```suggestion
        onPress={() => clipboardSetStringWithFeedback(props.otp.code)}
```